### PR TITLE
CXX-1936 add awaited to heartbeat events

### PR DESCRIPTION
**Pull request by kevinAlbs**
The C++ version of https://github.com/mongodb/mongo-c-driver/pull/621!

### CODE REVIEW

File: src/mongocxx/test/sdam-monitoring.cpp [hunk 1]
```diff
@@ -241,13 +258,20 @@ TEST_CASE("SDAM Monitoring", "[sdam_monitoring]") {
         open_and_close_client(test_uri, apm_opts);
         REQUIRE(heartbeat_started_events > 0);
         REQUIRE(heartbeat_succeeded_events > 0);
+        REQUIRE(started_awaited_called);
+        REQUIRE(succeeded_awaited_called);
```
Comment (kevinAlbs) [inline src/mongocxx/test/sdam-monitoring.cpp:262]: This does not test that awaited is ever true. But this does test we're calling the underlying libmongoc function.

File: src/mongocxx/events/heartbeat_failed_event.cpp [hunk 2]
```diff
@@ -49,6 +49,11 @@ std::uint16_t heartbeat_failed_event::port() const {
         ->port;
 }
 
+bool heartbeat_failed_event::awaited() const {
+    return libmongoc::apm_server_heartbeat_failed_get_awaited(
+        static_cast<const mongoc_apm_server_heartbeat_failed_t*>(_failed_event));
```
Comment (samantharitter) [inline src/mongocxx/events/heartbeat_failed_event.cpp:54]: spacing here looks different from other methods, clang format?

File: src/mongocxx/events/heartbeat_succeeded_event.hpp [hunk 3]
```diff
@@ -68,6 +68,12 @@ class MONGOCXX_API heartbeat_succeeded_event {
     ///
     std::uint16_t port() const;
 
+    ///
+    /// Returns a boolean indicating whether this heartbeat event is from an awaitable isMaster.
+    ///
+    /// @return A boolean.
```
[the remark this answers is not in the document]
Comment (kevinAlbs) [reply src/mongocxx/events/heartbeat_succeeded_event.hpp:74]: Gah, I keep forgetting this.

File: src/mongocxx/events/heartbeat_failed_event.hpp [hunk 4]
```diff
@@ -67,6 +67,12 @@ class MONGOCXX_API heartbeat_failed_event {
     ///
     std::uint16_t port() const;
 
+    ///
+    /// Returns a boolean indicating whether this heartbeat event is from an awaitable isMaster.
+    ///
+    /// @return A boolean.
```
[the remark this answers is not in the document]
Comment (kevinAlbs) [reply src/mongocxx/events/heartbeat_failed_event.hpp:73]: Done.

File: src/mongocxx/events/heartbeat_started_event.hpp [hunk 5]
```diff
@@ -53,6 +53,12 @@ class MONGOCXX_API heartbeat_started_event {
     ///
     std::uint16_t port() const;
 
+    ///
+    /// Returns a boolean indicating whether this heartbeat event is from an awaitable isMaster.
+    ///
+    /// @return A boolean.
```
[the remark this answers is not in the document]
Comment (kevinAlbs) [reply src/mongocxx/events/heartbeat_started_event.hpp:59]: Done.

File: src/mongocxx/events/heartbeat_failed_event.cpp (hunk 2 above)
Comment (kevinAlbs) [reply src/mongocxx/events/heartbeat_failed_event.cpp]: I ran clang format again to check, but that is how it formats it. I split these into separate lines for readability.

**Pull request merged**